File: 6.py
from itertools import imap
import sys


def square(x):
    return x ** 2


def sum_of_squares(numbers):
    return sum(imap(square, numbers))


def square_of_sums(numbers):
    return square(sum(numbers))


def main(x):
    seq = xrange(1, x + 1)
    return square_of_sums(seq) - sum_of_squares(seq)


if __name__ == '__main__':
    try:
        x = int(sys.argv[1])
    except (IndexError, ValueError):
        x = 100
    print main(x)

File: 6/6.go
package main

import "math"

func sumOfSquares(n float64) float64 {
	return n * (n + 1) * (2*n + 1) / 6
}

func squareOfSums(n float64) float64 {
	return math.Pow(n * (n + 1) / 2, 2)
}

func main() {
	x := 100.0
	println(int(squareOfSums(x) - sumOfSquares(x)))
}
